src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1291,15 +1299,6 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                         f"{candidate_file} as defined in `model_index.json` does not exist in {pretrained_model_name} and is not a module in 'diffusers/pipelines'."
                     )
 
-            if len(variant_filenames) == 0 and variant is not None:
```

Comment:
I didn't find the error from `download` still,

Reply:
Could you clarify what you mean?

Reply:
In any case, we do error out here, though https://github.com/huggingface/diffusers/pull/9253#discussion_r1766096902

Reply:
Or are you suggesting we should not remove this block of code? In that case, I think explained myself here already https://github.com/huggingface/diffusers/pull/9253/files#r1766067105 and I would appreciate some reasoning behind why that's not a good idea for my understanding.

Reply:
just put back the error here inside the download, no need to add it for local files because we have a similar check after `_identify_model_variants`

we want to add the error here because:
1. it errors out sooner
2. the `download` function as expected

Reply:
Makes sense. Thanks for explaining.

Reply:
Done in https://github.com/huggingface/diffusers/pull/9253/commits/25ac01f3455f9b4d18fe80057d247c791c872e3c. LMK.